sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/siprouting/SipRoutingAsyncClient.java

```diff
@@ -385,15 +394,45 @@ public Mono<Response<Void>> deleteTrunkWithResponse(String fqdn) {
         return Mono.just(new SimpleResponse<>(null, 200, null, null));
     }
 
-    private Mono<SipConfiguration> getSipConfiguration() {
+    /**
+    * Gets the list of routes matching the target phone number, ordered by priority.
+    * 
+    * @param targetPhoneNumber Phone number to test routing patterns against.
+    * @param sipConfigurationModel Sip configuration object to test with targetPhoneNumber.
+    * @return the list of routes matching the target phone number, ordered by priority along with {@link Response} on
+    * successful completion of {@link Mono}.
+    */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RoutesForNumber>> testRoutesWithNumberWithResponse(String targetPhoneNumber,
```

Comment:
We shouldn't expose a lower level model, let's map the api response to a List<SipTrunkRoute> to not introduce new public model.